docs/modules/ROOT/pages/discovery.adoc

```diff
@@ -0,0 +1,53 @@
+:clusterName: \{clusterName\}
+:namespace: \{namespace\}
+:metastorePort: 9083
+:roleGroup: \{roleGroup\}
+
+= Discovery
+
+The Stackable Operator for Apache Hive publishes a discovery https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.23/#configmap-v1-core[`ConfigMap`], which exposes a client configuration bundle that allows access to the Apache Hive cluster.
+
+The bundle includes a thrift connection string to access the Hive metastore service. This string may be used by other operators or tools to configure their products with access to Hive. This is limited to internal cluster access.
+
+== Example
+
+Given the following Hive cluster:
+
+[source,yaml,subs="normal,callouts"]
+----
+apiVersion: hive.stackable.tech/v1alpha1
+kind: HiveCluster
+metadata:
+  name: {clusterName} # <1>
+  namespace: {namespace} # <2>
+spec:
+  metastore:
+    roleGroups:
+      default: # <3>
+  [...]
+----
+<1> The name of the Hive cluster, which is also the name of the created discovery `ConfigMap`.
+<2> The namespace of the discovery `ConfigMap`.
+<3> A role group name of the metastore role.
+
+The resulting discovery `ConfigMap` is `{namespace}/{clusterName}`.
+
+== Contents
+
+The `{namespace}/{clusterName}` discovery `ConfigMap` contains the following fields where `{clusterName}` represents the name, `{namespace}` the namespace of the cluster and `{roleGroup}` a role group of the `metastore` role:
+
+`HIVE`::
+====
+Contains the thrift protocol connection string for the Hive metastore:
+
+[subs="attributes"]
+  thrift://{clusterName}-metastore-{roleGroup}-0.{clusterName}-metastore-{roleGroup}.{namespace}.svc.cluster.local:{metastorePort}
+
+or
+
+[subs="attributes"]
+  thrift://{clusterName}-metastore-{roleGroup}-0.{clusterName}-metastore-{roleGroup}.{namespace}.svc.cluster.local:{metastorePort}
+  thrift://{clusterName}-metastore-{roleGroup}-1.{clusterName}-metastore-{roleGroup}.{namespace}.svc.cluster.local:{metastorePort}
```

Comment:
Raising this here anyway, but this is more of an implementation question that we might want to raise as a separate issue...

Is this format for failover actually supported by Hive/Thrift clients? What are the consequences here if running in HA mode without upstream support? (Stale caches, data corruption, ???) Depending on the answer here, _do we want to support this mode of operation_?

Reply:
We do not have HA. These would be completely separated metastore instances afaik (there fore no caches or data corruption). I thought about limiting the replicas to 1 already but never introduced that. 

So yeah i would not support that for now.